derive/src/lib.rs

```diff
@@ -23,7 +31,43 @@ macro_rules! ident {
     };
 }
 
-fn expand_contract(input: LitStr) -> Result<TokenStream> {
+struct ContractArgs {
+    artifact_path: LitStr,
+    runtime_crate: Option<Ident>,
+}
+
+impl Parse for ContractArgs {
```

Comment:
Yes that would make sense.

Reply:
Yeah, it would have been better to split this up actually:
- one PR for each issue (5 total)
- one PR for extra changes required for getting the examples to run.